functions_framework_tool/templates/cloudevent/README.md

```diff
@@ -0,0 +1,118 @@
+# Basic CloudEvent example
+
+This example demonstrates writing a function to handle a CloudEvent.
+
+CloudEvent function handlers don't return a response to send to the event
+producter. They generally perform some work and print output for logging.
+
+The basic shape of the function handler looks like this:
+
+```dart
+@CloudFunction()
+void function(CloudEvent event, RequestContext context) {
+}
+```
+
+Or like this if it needs to perform work that will complete sometime in the
+future:
+
+```dart
+@CloudFunction()
+FutureOr<void> function(CloudEvent event, RequestContext context) async {
+}
+```
+
+The full code of the function for this example is shown below:
+
+lib/functions.dart
+
+```dart
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:functions_framework/functions_framework.dart';
+
+const _encoder = JsonEncoder.withIndent(' ');
+
+@CloudFunction()
+void function(CloudEvent event, RequestContext context) {
+  context.logger.info('event subject: ${event.subject}');
+  stderr.writeln(_encoder.convert(event));
+}
+```
+
+All the function does is log the source and subject of the CloudEvent that
+triggered it, and then prints out the entire event JSON object for informational
+purposes.
+
+## Generate project files
+
+The Dart `build_runner` tool generates `bin/server.dart`, the main entry point
+for the function server app, which invokes the function in `lib/functions.dart`.
+
+Run the `build_runner` tool, as shown here:
+
+```shell
+$ dart run build_runner build
```

Comment:
This will be a **lot** of edits to make in one pass -- includes everything under `docs` and `examples` (which are out of this PR's scope. Opened a ticket to make a full pass through the repo: #135